src/CoreWCF.NetTcp/tests/TransportWithMessageCredentialNetTCPTest.cs

```diff
@@ -0,0 +1,144 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Runtime.ConstrainedExecution;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.ServiceModel.Channels;
+using CoreWCF.Configuration;
+using CoreWCF.IdentityModel.Selectors;
+using CoreWCF.Security;
+using Helpers;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace CoreWCF.NetTcp.Tests
+{
+    public class TransportWithMessageCredentialNetTCPTest
+    {
+        private readonly ITestOutputHelper _output;
+
+        public TransportWithMessageCredentialNetTCPTest(ITestOutputHelper output)
+        {
+            _output = output;
+        }
+
+        [Fact ]
+        public void BasicUserNameAuth()
+        {
```

Comment:
Use `[Theory]` instead of `[Fact]` and you can have parameterized tests. [Here's a good blog post](https://hamidmosalla.com/2017/02/25/xunit-theory-working-with-inlinedata-memberdata-classdata/) with brief examples. Using InlineData here would probably be the right option as there's only two variants. If you were to stick to this method, the casing on runTest is wrong. Also _output is an instance field so doesn't need to be passed. Something like:
```c#
[Theory]
[InlineData(false, "testuser@corewcf"]
[InlineData(true, "randomuser@corewcf"]
public void BasicUserNameAuth(bool isError, string userName)
{
    ///
}
```